docs/current_docs/manuals/developer/ide-integration.mdx

```diff
@@ -43,69 +43,151 @@ echo go.work.sum >> .gitignore
-poetry add -e --group=dev ./sdk
+echo 3.11 > .python-version
 ```
+
+:::
+
 </TabItem>
-<TabItem value="Hatch">
+<TabItem value="pip">
 
-Install the Python SDK from the generated `./sdk` directory, as shown below:
+Make sure you have the same Python version that Dagger uses first. By default it's Python 3.12, unless it's overriden in `pyproject.toml` or in `.python-version`.
 
 ```shell
-pip install -e ./sdk
+python -m venv .venv
+source .venv/bin/activate
+python -m pip install -r requirements.lock -e ./sdk -e .
 ```
 
-Update `pyproject.toml` to [allow direct references](https://hatch.pypa.io/latest/config/metadata/#allowing-direct-references), and add a [local](https://hatch.pypa.io/latest/config/dependency/#local) dependency:
+</TabItem>
+<TabItem value="hatch">
+
+First, update `pyproject.toml` to [allow direct references](https://hatch.pypa.io/latest/config/metadata/#allowing-direct-references), and add the [local](https://hatch.pypa.io/latest/config/dependency/#local) SDK library (`dagger-io`) as a  **dev** dependency:
 
 ```toml
 [project]
 name = "main"
 version = "0.0.0"
 
+[tool.hatch.metadata]
+allow-direct-references = true
+
 [tool.hatch.envs.dev]
 dependencies = [
   "dagger-io @ {root:uri}/sdk",
 ]
+```
 
-[tool.hatch.metadata]
-allow-direct-references = true
+Then, open the editor with the *dev* [Hatch environment](https://hatch.pypa.io/latest/config/environment/overview/):
+
+```shell
+hatch run dev:vim .
+```
+
+:::note
+Hatch manages the project environment in a central location by default instead of `.venv`.
+:::
+
+</TabItem>
+<TabItem value="poetry">
+
+First, add `./sdk` as a development dependency:
+
+```shell
+poetry add --group=dev -e ./sdk
 ```
 
-Confirm it was installed:
+Then, open the editor with:
 
 ```shell
-hatch run dev:pip list
+poetry run vim .
 ```
+
+:::note
+Poetry manages the project environment in a central location instead of `.venv`.
+:::
+
 </TabItem>
 </Tabs>
 
-For other package managers, check their documentation on how to accomplish the same.
-
 :::tip
 If you place the virtual environment (`.venv`) inside the module, don't forget to add it to `.gitignore` and to `"exclude"` in `dagger.json` to avoid uploading those files to the runtime container unnecessarily.
 :::
 
-:::note
-It's important to install the SDK in editable mode (the `-e` in the `pip install` and `poetry add` commands) so you don't have to reinstall it after a `dagger develop`. However, it's always recommended to reinstall the SDK after upgrading Dagger to cover possible dependency changes.
-:::
+## Visual Studio Code
+
+The simplest way to get [autocompletions and IntelliSense](https://code.visualstudio.com/docs/python/editing#_autocomplete-and-intellisense) working is to open it with the directory hat has `.venv` (next to `.pyproject.toml`), because VS Code picks it up automatically when it's in the root of the [VS Code workspace](https://code.visualstudio.com/docs/python/python-tutorial#_start-vs-code-in-a-workspace-folder):
```

Comment:
```suggestion
The simplest way to get [autocompletions and IntelliSense](https://code.visualstudio.com/docs/python/editing#_autocomplete-and-intellisense) working is to open it with the directory that has `.venv` (next to `.pyproject.toml`), because the IDE picks it up automatically when it's in the root of the [workspace](https://code.visualstudio.com/docs/python/python-tutorial#_start-vs-code-in-a-workspace-folder):
```